src/core/objective.jl

```diff
@@ -273,6 +273,33 @@ function _objective_mc_min_fuel_cost_polynomial_linquad(pm::AbstractUnbalancedPo
 end
 
 
+"gen connections adaptation of min fuel cost polynomial linquad objective"
+function _objective_mc_min_fuel_cost_polynomial_linquad(pm::ExplicitNeutralModels; report::Bool=true)
+    gen_cost = Dict()
+    for (n, nw_ref) in nws(pm)
+        for (i,gen) in nw_ref[:gen]
+            pg = sum( var(pm, n, :pg, i) )
+
+            if length(gen["cost"]) == 1
+                gen_cost[(n,i)] = gen["cost"][1]
+            elseif length(gen["cost"]) == 2
+                gen_cost[(n,i)] = gen["cost"][1]*pg + gen["cost"][2]
+            elseif length(gen["cost"]) == 3
+                gen_cost[(n,i)] = gen["cost"][1]*pg^2 + gen["cost"][2]*pg + gen["cost"][3]
+            else
+                gen_cost[(n,i)] = 0.0
+            end
+        end
+    end
+
+    return JuMP.@objective(pm.model, Min,
+        sum(
+            sum( gen_cost[(n,i)] for (i,gen) in nw_ref[:gen] )
+        for (n, nw_ref) in nws(pm))
+    )
+end
```

Comment:
I merged all into a single function, which inspects whether `pg` contains any NL expressions and then acts accordingly. Typically we prefer doing this by overloading, but that does not work here because along the type hierarchy there is not a single branch which has NL expressions; i.e., the quadratic IVR does not have these, whilst the parent IVR has NL expressions, and this leads to a lot of code duplication. The change I made now prevents this.